pyexcel/internal/sheets/matrix.py

```diff
@@ -82,18 +82,18 @@ def cell_value(self, row, column, new_value=None):
         :param int column: column index which starts from 0
         :param any new_value: new value if this is to set the value
         """
-        if row < self.number_of_rows() and column < self.number_of_columns():
-            if new_value is None:
-                # get
+        fit = row < self.number_of_rows() and column < self.number_of_columns()
+        if new_value is None:
+            if fit:
                 return self.__array[row][column]
             else:
-                # set
-                self.__array[row][column] = new_value
-        else:
-            if new_value is None:
                 raise IndexError("Index out of range")
-            else:
-                self.paste((row, column), [[new_value]])
+        else:
+            if not fit:
+                width, array = uniform(self.__array, row+1, column+1)
```

Comment:
I think here, it is supposed that row > number of rows and column > number of rows.

what happens if: 

1. row < number of rows but column > number of rows
2. row > number of rows but column < number of rows

Reply:
it is good to add two tests to test above two conditions

Reply:
i added those two edge cases as test, but to me it looks pretty clear:

```
fit = row < row_num and column < column_num
row < row_num, column > column_num => True and False => fit = false
row > row_num, column < coulmn_num => False and True => fit = false
```